airflow/jobs.py

```diff
@@ -1026,6 +1026,21 @@ def _change_state_for_tis_without_dagrun(self,
             )
 
     @provide_session
+    def __get_task_concurrency_map(self, session=None):
```

Comment:
There's no reason why we call this from a static context. It doesn't quite belong in the task class because it creates a map which is specific to the scheduler.

Reply:
self isn't being used so it's a static method, I think most linters would complain about this.

Reply:
Not overrideable/inheritable though. Not sure if it matters here but that's why I keep it in class context.